app/src/org/commcare/views/widgets/GregorianWidget.java

```diff
@@ -0,0 +1,136 @@
+package org.commcare.views.widgets;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.commcare.dalvik.R;
+import org.commcare.utils.UniversalDate;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.joda.time.DateTime;
+
+import java.util.Calendar;
+
+/**
+ * Created by Saumya on 5/27/2016.
+ * Uses the increment/decrement button structure to input Gregorian dates
+ */
+public class GregorianWidget extends AbstractUniversalDateWidget {
+
+    private EditText dayTxt;
+    private EditText monthTxt;
+    private EditText yearTxt;
+    private Calendar myCal;
+
+    private String[] myMonths;
+
+    public GregorianWidget(Context context, FormEntryPrompt prompt){
+        super(context, prompt); //Adds all the stuff for abstract date widget
+        myCal = Calendar.getInstance();
+
+
+        //TODO: Validate input from text fields: Use a TextWatcher for this, use myCal to validate days in month
+        //TODO: Update month array pointer based on text entered..check to see how super does this though
+    }
+
+    @Override
+    protected void initText(){
+        dayTxt = (EditText)findViewById(R.id.daytxt);

```

Comment:
My IDE is complaining about this being a TextView - this is because, confusingly, Android resource id's aren't domained by the file so if you have +id/daytxt in gregorian_date_widget.xml and universal_date_widget.xml then that reference will become ambiguous. So you'll need new IDs in your layout file